jupyterhub_singleuser_profiles/profiles.py

```diff
@@ -226,7 +248,24 @@ def apply_gpu_config(self, gpu_mode, gpu_count, pod):
         if gpu_mode == self.GPU_MODE_PRIVILEGED:
           pod.spec.security_context.privileged = True
 
-    return pod
+      if gpuTolerations:
+        # We currently do not have a way to select the type of GPU in the notebook spawner
+        # Our workaround for the time being is to apply all possible gpu tolerations
+        if gpuType == "ALL":
+          for gpuToleration in gpuTolerations:
+            node_tolerations.extend(gpuToleration.get('node_tolerations', []))
+        else:
+          for gpuToleration in gpuTolerations:
+            if gpuType == gpuToleration.get('type'):
+              node_tolerations.extend(gpuToleration.get('node_tolerations', []))
+              break
+
+    node_tolerations.extend(profile.get('node_tolerations', []))
+    node_affinity = profile.get('node_affinity', {})
```

Comment:
This looks really weird here because the method is to handle GPUs and we are suddenly applying stuff from profile potentially unrelatied to GPUs. Since we only have `node_tolerations` in GPUs config, I would suggest you call the `apply_pod_schedulers` twice - once here with the GPU tolerations and then again from `apply_pod_profile`.

You need to make sure the `apply_pod_schedulers` does not overwrite but extends.

WDYT?